tests/integration/tabular/test_base.py

```diff
@@ -214,3 +216,82 @@ def test_conditional_sampling_constraint_uses_columns_model(gm_mock):
     pd.testing.assert_series_equal(sampled_data['age'], expected_ages)
     pd.testing.assert_series_equal(sampled_data['state'], expected_states)
     assert all(c in ('SF', 'LA') for c in sampled_data['city'])
+
+
+@patch('sdv.tabular.copulas.copulas.multivariate.GaussianMultivariate',
+       spec_set=GaussianMultivariate)
+def test_conditional_sampling_constraint_uses_columns_model_reject_sampling(gm_mock):
+    """Test that the ``sample`` method handles constraints with conditions.
+
+    The ``sample`` method is expected to properly apply constraint
+    transformations by sampling the missing columns for the constraint
+    if ``fit_columns_model`` is True. All values sampled by the column
+    model should be valid because reject sampling is used on any that aren't.
+
+    Setup:
+    - The model is being passed a ``GreaterThan`` constraint and then
```

Comment:
This description of the test is great! However, the `Setup` section should be more about "What needs to be set up for this test to happen". So, in this case, rather than explaining what is expected to happen, I would mention that the setup involves a mock of the GM that will sample rows that do not satisfy the constraint to force the reject sampling to be used.

Reply:
The reason for this is that the model being mocked is not the one in the constraint but the one in sdv. I didn't mock the constraint's `_columns_model` since it is an integration test. I only mocked the other model so that we don't have to wait for it to fit

Reply:
Update: I mocked the column model to force reject sampling